src/Resolvers/AbstractDataResolver.php

```diff
@@ -35,11 +38,34 @@ public function validate(array $data): array {
      * {@inheritDoc}
      */
     final public function resolve(array $data, array $params = []) {
-        $result = $this->validate($data);
-        $result = $this->resolveInternal($result, $params);
+        global $layoutCacheNode;
```

Comment:
Why are we using a global variable?

Reply:
This is explained in the description. Class is not a singleton and making the var global allows it to be shared with other instances of resolve() in the same request.